lfsapi/creds.go

```diff
@@ -90,76 +90,72 @@ type AskPassCredentialHelper struct {
 // The ASKPASS program is only queried if a credential was not already
 // provided, i.e. through the git URL
 func (a *AskPassCredentialHelper) Fill(what Creds) (Creds, error) {
-	var user bytes.Buffer
-	var pass bytes.Buffer
-	var err bytes.Buffer
-
-	var username string
-	var password string
-
 	u := &url.URL{
 		Scheme: what["protocol"],
 		Host:   what["host"],
 		Path:   what["path"],
 	}
 
+	creds := make(Creds)
+
 	if givenUser, ok := what["username"]; ok {
```

Comment:
This pattern roughly breaks down for me as:

1. See if `what` has a given key, ex. "username".
2. If so, copy it to `creds`.
3. If not, invoke `getFromProgram` with some mapping of `username` -> `Username`.
4. If the above is OK, copy that into `creds`.
5. If not, report an error.

If this is the same for `username` and `password`, could we extract another function that captures the above flow?